lib/facility.rb

```diff
@@ -1,14 +1,63 @@
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :permit
+  attr_accessor :collected_fees, :registration_date, :plate, :data
+                
+
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @data = data
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
+    @services
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') && registrant.permit?
+      true
+    else
+      false
+    end
+  end
+
+  def administer_road_test(registrant)
+    if @services.include?('Road Test') && registrant.age >15
+      true
+    else
+      false
+    end
+  end
+
+  def renew_drivers_license(registrant)
+    if @services.include?('Renew License') && registrant.administer_road_test
+      true
+    else
+      false
+    end
+  end
+
+  def register_vehicle(vehicle)
+      if vehicle.antique?
+        @collected_fees += 25
```

Comment:
Collecting fees could be put into a smaller helper method! Promote SRP by having functions do just one thing really well - even if it includes calling on helper functions!